vcomponent/src/vcomponent_hdmi_cec_device.c

```diff
@@ -0,0 +1,465 @@
+/*
+ * If not stated otherwise in this file or this component's LICENSE file the
+ * following copyright and licenses apply:
+ *
+ * Copyright 2016 RDK Management
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+#include <string.h>
+#include <stdlib.h>
+#include <stdbool.h>
+#include <assert.h>
+
+#include "vcomponent_hdmi_cec_device.h"
+
+
+const static strVal_t gDIStrVal [] = {
+  { "TV", (int) DEVICE_TYPE_TV },
+  { "PlaybackDevice", (int) DEVICE_TYPE_PLAYBACK },
+  { "AudioSystem" , (int) DEVICE_TYPE_AUDIO_SYSTEM },
+  { "RecordingDevice", (int) DEVICE_TYPE_RECORDER },
+  { "Tuner", (int) DEVICE_TYPE_TUNER },
+  { "Reserved",  (int) DEVICE_TYPE_RESERVED },
+  { "FreeUse", (int) DEVICE_TYPE_FREEUSE },
+  { "Unregistered", (int) DEVICE_TYPE_UNREGISTERED },
+  { "Unknown", (int) DEVICE_TYPE_UNKNOWN }
+};
+
+const static strVal_t gVCStrVal [] = {
+  {"TOSHIBA", VENDOR_CODE_TOSHIBA},
+  {"SAMSUNG", VENDOR_CODE_SAMSUNG},
+  {"DENON", VENDOR_CODE_DENON},
+  {"MARANTZ", VENDOR_CODE_MARANTZ},
+  {"LOEWE", VENDOR_CODE_LOEWE},
+  {"ONKYO", VENDOR_CODE_ONKYO},
+  {"MEDION", VENDOR_CODE_MEDION},
+  {"TOSHIBA2", VENDOR_CODE_TOSHIBA},
+  {"APPLE", VENDOR_CODE_APPLE},
+  {"HARMAN_KARDON2", VENDOR_CODE_HARMAN_KARDON2},
+  {"GOOGLE", VENDOR_CODE_GOOGLE},
+  {"AKAI", VENDOR_CODE_AKAI},
+  {"AOC", VENDOR_CODE_AOC},
+  {"PANASONIC", VENDOR_CODE_PANASONIC},
+  {"PHILIPS", VENDOR_CODE_PHILIPS},
+  {"DAEWOO", VENDOR_CODE_DAEWOO},
+  {"YAMAHA", VENDOR_CODE_YAMAHA},
+  {"GRUNDIG", VENDOR_CODE_GRUNDIG},
+  {"PIONEER", VENDOR_CODE_PIONEER},
+  {"LG", VENDOR_CODE_LG},
+  {"SHARP", VENDOR_CODE_SHARP},
+  {"SONY", VENDOR_CODE_SONY},
+  {"BROADCOM", VENDOR_CODE_BROADCOM},
+  {"SHARP2", VENDOR_CODE_SHARP2},
+  {"VIZIO", VENDOR_CODE_VIZIO},
+  {"BENQ", VENDOR_CODE_BENQ},
+  {"HARMAN_KARDON", VENDOR_CODE_HARMAN_KARDON},
+  {"UNKNOWN", VENDOR_CODE_UNKNOWN},
+  };
+
+const static strVal_t gPSStrVal [] = {
+  { "on", (int)POWER_STATUS_ON  },
+  { "off", (int)POWER_STATUS_OFF  },
+  { "standby", (int)POWER_STATUS_STANDBY },
+  { "unknown", (int)POWER_STATUS_UNKNOWN }
+};
+
+/* Load the device info into the passed in vCHdmiCec_device_info_t*
+* prefix can be 
+*   "hdmicec/device_map/0"
+*   "hdmicec/device_map/0/children/0"
+*   "hdmicec/device_map/0/children/1"
+*   "hdmicec/device_map/0/children/1/children/0"
+*/
+void LoadDeviceInfo (ut_kvp_instance_t* instance, char* prefix, struct vCHdmiCec_device_info_t* device)
```

Comment:
internal functions need to be static.
external functions need to be prefixed by the filename

`vcomponent_hdmi_cec_device_`, so maybe better to change the filename the add the prefix, so you know when your reading the code which module it's talking about.